### ISSUE TITLE
Should we track the version of `kind` CLI

### ISSUE THREAD

**Issue by pangruoran** [Kong/kubernetes-ingress-controller#5131]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Some incompatibilities exist between different kind versions and the old version can't work with Kubernetes v1.28.

We have not encountered any issues because the GitHub Actions environment has already upgraded kind to the latest version by default.

https://github.com/actions/runner-images/pull/7747

In PR #5088, I bumped the version of `kind` CLI.

### Proposed Solution

* track the version of the `kind` CLI in the `.github/test_dependencies.yaml` file

OR

* remove the installation logic of `kind` the our script.

### Additional information

>  Let's merge this @pangruoran to unblock you. I suggest to create a follow up issue to discuss alternatives (we shouldn't have the version hardcoded in the script like that. @programmer04 is right that on CI `kind` is preinstalled but [the makefile target](https://github.com/Kong/kubernetes-ingress-controller/blob/9b81665dd679539b8ce4298ea4b3cad5aec5b00c/Makefile#L409-L411) that uses this script is used in other places (other makefiles targets) and it would be nice to have this (kind version) defined in 1 place and not installed in a script which is meant to "check".

_Originally posted by @pmalek in https://github.com/Kong/kubernetes-ingress-controller/pull/5088#discussion_r1386268950_
